lib/src/config/dictionaries.rs

```diff
@@ -0,0 +1,189 @@
+mod deserialization {
+
+    use {
+        super::{DictionariesConfig, Dictionary, DictionaryName},
+        crate::error::{DictionaryConfigError, FastlyConfigError},
+        std::{convert::TryFrom, convert::TryInto, fs},
+        toml::value::{Table, Value},
+        tracing::{event, Level},
+    };
+
+    /// Helper function for converting a TOML [`Value`] into a [`Table`].
+    ///
+    /// This function checks that a value is a [`Value::Table`] variant and returns the underlying
+    /// [`Table`], or returns an error if the given value was not of the right type — e.g., a
+    /// [`Boolean`][Value::Boolean] or a [`String`][Value::String]).
+    fn into_table(value: Value) -> Result<Table, DictionaryConfigError> {
+        match value {
+            Value::Table(table) => Ok(table),
+            _ => Err(DictionaryConfigError::InvalidEntryType),
+        }
+    }
+
+    /// Return an [`DictionaryConfigError::UnrecognizedKey`] error if any unrecognized keys are found.
+    ///
+    /// This should be called after we have removed and validated the keys we expect in a [`Table`].
+    fn check_for_unrecognized_keys(table: &Table) -> Result<(), DictionaryConfigError> {
+        if let Some(key) = table.keys().next() {
+            // While other keys might still exist, we can at least return a helpful error including
+            // the name of *one* unrecognized keys we found.
+            Err(DictionaryConfigError::UnrecognizedKey(key.to_owned()))
+        } else {
+            Ok(())
+        }
+    }
+
+    impl TryFrom<Table> for DictionariesConfig {
+        type Error = FastlyConfigError;
+        fn try_from(toml: Table) -> Result<Self, Self::Error> {
+            /// Process a backend's definitions, or return a [`FastlyConfigError`].
+            fn process_entry(
+                (name, defs): (String, Value),
+            ) -> Result<(DictionaryName, Dictionary), FastlyConfigError> {
+                into_table(defs)
+                    .and_then(|mut toml| {
+                        let file = toml
+                            .remove("file")
+                            .ok_or(DictionaryConfigError::MissingFile)
+                            .and_then(|file| match file {
+                                Value::String(file) => {
+                                    if file.is_empty() {
+                                        Err(DictionaryConfigError::EmptyFileEntry)
+                                    } else {
+                                        Ok(file.into())
+                                    }
+                                }
+                                _ => Err(DictionaryConfigError::InvalidFileEntry),
+                            })?;
+                        check_for_unrecognized_keys(&toml)?;
+                        let dictionary_max_len: usize = 1000;
+                        let dictionary_item_key_max_len: usize = 256;
+                        let dictionary_item_value_max_len: usize = 8000;
```

Comment:
Could we pull these out into a dedicated `limits.rs` file? We're going to want to build more awareness of C@E limits in Viceroy like this, and having them in one place will make life easier for keeping them up to date.

Reply:
Done in [`b75566d` (#61)](https://github.com/fastly/Viceroy/pull/61/commits/b75566d5e4e78b7ea8b1e91a409d0aca15f2295a) 👍 

Should these also go in the fastly-shared crate?